### PR TITLE
Standardization of Lanth-Bot Chemical Recipes

### PR DESCRIPTION
**Pull request by GlodBlock**

### CODE REVIEW

File: src/main/java/com/elisis/gtnhlanth/loader/BotRecipes.java [hunk 1]
```diff
@@ -75,53 +75,58 @@ public static void addGTRecipe() {
...
 
 //        ItemStack WO3Fe = TungstenSteelOxide.get(dust, 2);
 //        GT_Values.RA.addMixerRecipe(WO3, Materials.Steel.getDust(1), null, null, null, null,
 //                WO3Fe, 100, 1920);
 
-        GT_Values.RA.addBlastRecipe(WO3,C2,Materials.Hydrogen.getGas(3000),
-                null,Materials.Tungsten.getDust(1),null,100,1920,1000);
+        //WO3 + 6H = W + 3H2O
+        GT_Values.RA.addBlastRecipe(WO3,C2,Materials.Hydrogen.getGas(6000),
+                Materials.Water.getFluid(3000),Materials.Tungsten.getDust(1),null,100,1920,1000);
```
Comment (miozune) [inline src/main/java/com/elisis/gtnhlanth/loader/BotRecipes.java]: I don't think EBF can output liquid form of water. Either output steam or nothing, just like Sodium Ruthenate

File: src/main/java/com/elisis/gtnhlanth/loader/BotRecipes.java [hunk 2]
```diff
@@ -139,24 +144,31 @@ public static void addGTRecipe() {
...
 
+        //C3H5ClO2 + 2NH3 = C3H7O2N + NH4Cl
         GT_Values.RA.addChemicalRecipe(Ethylchloroformate.get(cell, 1), C2, Materials.Ammonia.getGas(2000),
                 WerkstoffLoader.AmmoniumChloride.getFluidOrGas(1000), Ethylcarbamate.get(cell, 1), 200, 120);
 
+        //C3H7O2N + HNO3 = C3H6N2O4 + H2O
         GT_Values.RA.addChemicalRecipe(Ethylcarbamate.get(cell, 1), C2, Materials.NitricAcid.getFluid(1000),
                 Materials.Water.getFluid(1000), EthylNnitrocarbamate.get(cell, 1), 40, 1024);
 
+        //C3H6N2O4 + NH3 = C3H10N3O4
```
Comment (miozune) [inline src/main/java/com/elisis/gtnhlanth/loader/BotRecipes.java]: Ammonium N-nitrourethane is C3H9N3O4, so we need to adjust tooltip
https://en.wikipedia.org/wiki/Ammonium_dinitramide

File: src/main/java/com/elisis/gtnhlanth/loader/BotRecipes.java [hunk 3]
```diff
@@ -139,24 +144,31 @@ public static void addGTRecipe() {
...
         GT_Values.RA.addChemicalRecipe(EthylNnitrocarbamate.get(cell, 1), C2, Materials.Ammonia.getGas(1000),
                 null, AmmoniumNnitrourethane.get(cell, 1), 40, 1920);
 
-        GT_Values.RA.addChemicalRecipe(AmmoniumNnitrourethane.get(cell, 1), DinitrogenPentoxide.get(dust, 1), null, null,
-                EthylDinitrocarbamate.get(cell, 1), AmmoniumNitrate.get(dust, 1), 200, 480);
+        //C3H10N3O4 + N2O5 = C3H5N3O6 + N2H4O3 + H
+        GT_Values.RA.addChemicalRecipe(AmmoniumNnitrourethane.get(cell, 1), DinitrogenPentoxide.get(dust, 7), null, Materials.Hydrogen.getGas(1000),
+                EthylDinitrocarbamate.get(cell, 1), AmmoniumNitrate.get(dust, 9), 200, 480);
```
Comment (miozune) [inline src/main/java/com/elisis/gtnhlanth/loader/BotRecipes.java:166]: And hydrogen output is not needed as said above

File: src/main/java/com/elisis/gtnhlanth/loader/BotRecipes.java [hunk 4]
```diff
@@ -168,31 +180,37 @@ public static void addGTRecipe() {
...
         );
 
-        GT_Values.RA.addChemicalRecipe(WerkstoffLoader.SodiumNitrate.get(dust, 4), C2, Materials.SulfuricAcid.getFluid(2000),
-                Materials.NitricAcid.getFluid(2000), WerkstoffLoader.Sodiumsulfate.get(dust, 1), 200, 120);
+        //2NaNO3 + H2SO4 = 2HNO3 + Na2SO4
+        GT_Values.RA.addChemicalRecipe(WerkstoffLoader.SodiumNitrate.get(dust, 10), C2, Materials.SulfuricAcid.getFluid(1000),
+                Materials.NitricAcid.getFluid(2000), WerkstoffLoader.Sodiumsulfate.get(dust, 7), 200, 120);
 
+        //N2H4O3 + NaOH + H =H2SO4= NH3 + HNO3 + Na + H2O
         GT_Values.RA.addMultiblockChemicalRecipe(
```
Comment (miozune) [inline src/main/java/com/elisis/gtnhlanth/loader/BotRecipes.java:215]: We need to change these values according to equation

File: src/main/java/com/elisis/gtnhlanth/loader/BotRecipes.java [hunk 5]
```diff
@@ -289,26 +312,30 @@ public static void addGTRecipe() {
 
         //unsimetrical hydazine
 
-        GT_Values.RA.addCrackingRecipe(2,Materials.Propane.getGas(1000),Materials.NitricAcid.getFluid(1000),
-                Nitromethane.getFluidOrGas(1000),300,1920);
+        //2HNO3 + C3H8 = 2CH3NO2 + 2H2O + C
+        GT_Values.RA.addCrackingRecipe(2,Materials.Propane.getGas(1000),Materials.NitricAcid.getFluid(2000),
+                Nitromethane.getFluidOrGas(2000),300,1920);
 
+        //HF + BF3 + CH3NO2 + HNO3 + H = CH4O + H2O + NO + NO2BF4
```
Comment (miozune) [inline src/main/java/com/elisis/gtnhlanth/loader/BotRecipes.java]: It seems that nitromethane itself is not involved in the equation. I'm not sure if it's some kind of catalyst, but at least adding H to reaction doesn't seem to be fine.
http://www.orgsyn.org/demo.aspx?prep=CV5P0480
https://pubs.acs.org/doi/pdf/10.1021/jo00237a043

File: src/main/java/com/elisis/gtnhlanth/loader/BotRecipes.java [hunk 6]
```diff
@@ -318,29 +345,41 @@ public static void addGTRecipe() {
                 20,30_720
         );
 
+        //B2O3 + 6HF = 2BF3 + 3H2O
         cells.stackSize = 3;
-        GT_Values.RA.addChemicalRecipe(BoronTrioxide.get(dust,1),cells,Materials.HydrofluoricAcid.getFluid(6000),
+        GT_Values.RA.addChemicalRecipe(BoronTrioxide.get(dust,5),cells,Materials.HydrofluoricAcid.getFluid(6000),
                 BoronTrifluoride.getFluidOrGas(2000),Materials.Water.getCells(3),50,480);
 
         GT_Values.RA.addChemicalRecipe(Materials.Borax.getDust(23),C2,Materials.SulfuricAcid.getFluid(1000),
```
Comment (miozune) [inline src/main/java/com/elisis/gtnhlanth/loader/BotRecipes.java:359]: Na2B4O7(H2O)10 + H2SO4 -> 2B2O3 + Na2SO4 + 11H2O (recipe is fine, just add comment)

File: src/main/java/com/elisis/gtnhlanth/loader/RecipeLoader.java [hunk 7]
```diff
@@ -24,148 +24,155 @@ public class RecipeLoader {
...
+                null,
+                new int[] {10000, 10000},
+                600,
                 1920
             );
-        
+
         //Ammonium Nitrate
+        //HNO3 + NH3 = NH4NO3
         GT_Values.RA.addChemicalRecipe(
```
Comment (miozune) [inline src/main/java/com/elisis/gtnhlanth/loader/RecipeLoader.java]: this does not generate LCR recipe, not sure why though

File: src/main/java/com/elisis/gtnhlanth/loader/RecipeLoader.java [hunk 8]
```diff
@@ -969,12 +1005,10 @@ public static void loadLanthanideRecipes() {
...
         		480
         		);
-        
-        
-        
+
         GT_Values.RA.addBlastRecipe(
         		WerkstoffMaterialPool.FluorinatedSamaricConcentrate.get(OrePrefixes.dust, 2),
-        		Materials.Calcium.getDust(3),
+        		Materials.Calcium.getDust(2),
```
Comment (miozune) [inline src/main/java/com/elisis/gtnhlanth/loader/RecipeLoader.java]: 1x calcium

File: src/main/java/com/elisis/gtnhlanth/loader/BotRecipes.java [hunk 9]
```diff
@@ -318,29 +345,41 @@ public static void addGTRecipe() {
...
-                null,WerkstoffLoader.Sodiumsulfate.get(dust,1),BoronTrioxide.get(dust,2),400,1920);
+                null,WerkstoffLoader.Sodiumsulfate.get(dust,7),BoronTrioxide.get(dust,10),400,1920);
 
+        //NH4BF4 = NH3 + BF4 + H
```
Comment (miozune) [inline src/main/java/com/elisis/gtnhlanth/loader/BotRecipes.java]: Tetrafluoroborate is ion (BF4-), so it should be like this:
- NH4BF4 -> HBF4 + NH3
- HBF4 + NaOH -> NaBF4 + H2O

Can you rename `Tetrafluoroborate` to `Tetrafluoroboric Acid` and adjust recipes?

**Review by miozune — COMMENTED** [review]
- 9x ammonium nitrate input for nitrogenated samarium-terbium mixture dust
- 5x tricalcium phosphate output

some recipes for PTMEG are not standardized, but I cannot find enough documents, so they can be left untouched I guess

File: src/main/java/com/elisis/gtnhlanth/loader/BotRecipes.java (hunk 5 above)
Comment (GlodBlock) [reply src/main/java/com/elisis/gtnhlanth/loader/BotRecipes.java]: i make carbon works as reductant. but in fact it is the combintion of two reaction so the nitromethane looks like something that provide energy for the reaction. it also can be the turbine fuel i think

File: src/main/java/com/elisis/gtnhlanth/loader/BotRecipes.java [hunk 10]
```diff
@@ -117,7 +118,7 @@ public static void addGTRecipe() {
 
         //WO3 + 6H = W + 3H2O
         GT_Values.RA.addBlastRecipe(WO3,C2,Materials.Hydrogen.getGas(6000),
-                Materials.Water.getFluid(3000),Materials.Tungsten.getDust(1),null,100,1920,1000);
+            GT_ModHandler.getSteam(6000),Materials.Tungsten.getDust(1),null,100,1920,1000);
```
Comment (miozune) [inline src/main/java/com/elisis/gtnhlanth/loader/BotRecipes.java]: Why 6000mB?

File: src/main/java/com/elisis/gtnhlanth/loader/BotRecipes.java [hunk 11]
```diff
@@ -118,7 +118,7 @@ public static void addGTRecipe() {
 
         //WO3 + 6H = W + 3H2O
         GT_Values.RA.addBlastRecipe(WO3,C2,Materials.Hydrogen.getGas(6000),
-            GT_ModHandler.getSteam(6000),Materials.Tungsten.getDust(1),null,100,1920,1000);
+            GT_ModHandler.getSteam(480000),Materials.Tungsten.getDust(1),null,100,1920,1000);
```
Comment (miozune) [inline src/main/java/com/elisis/gtnhlanth/loader/BotRecipes.java]: I'm not sure but isn't 3000mB enough?

Comment (GlodBlock) [reply src/main/java/com/elisis/gtnhlanth/loader/BotRecipes.java]: 1mb water = 160mb steam

Comment (miozune) [reply src/main/java/com/elisis/gtnhlanth/loader/BotRecipes.java]: yeah but other recipes (ruthenium / potassium) output "small" amount of steam (1000mB / 16000mB)

Comment (GlodBlock) [reply src/main/java/com/elisis/gtnhlanth/loader/BotRecipes.java]: fine

**Review by miozune — APPROVED** [review]
Thanks. If we made 160x steam as standard, we'll need 320B of steam to make 1B of ethanol, which requires LuV Hatches.

**Pull request merged**